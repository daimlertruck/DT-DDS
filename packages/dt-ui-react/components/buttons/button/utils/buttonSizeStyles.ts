import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
