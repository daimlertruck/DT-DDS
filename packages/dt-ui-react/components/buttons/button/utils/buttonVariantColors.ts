import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
  TextColors,
} from '../constants/types';

interface ColorKeys {
  color: string;
  bgColor: string;
  hoverBgColor?: string;
  hoverColor?: string;
}

export const buttonVariantColors = (
  color: ContainedColors | OutlinedColors,
  theme: Theme
): Record<ButtonVariant, ColorKeys> => ({
  contained: {
    color: theme.colors.alpha.white,
    bgColor: theme.palette.alpha[color],
    hoverBgColor: theme.palette.alpha[`${color}Dark_300` as ContainedColors],
  },
  outlined: {
    color:
      color === 'neutral'
        ? theme.colors.alpha.gray_200
        : theme.palette.alpha[color],
    bgColor: theme.colors.alpha.white,
    hoverBgColor: theme.palette.alpha[`${color}Light_300` as OutlinedColors],
  },
  text: {
    color: theme.palette.alpha[color],
    bgColor: 'transparent',
    hoverColor: theme.palette.alpha[`${color as TextColors}Dark_300`],
  },
});
