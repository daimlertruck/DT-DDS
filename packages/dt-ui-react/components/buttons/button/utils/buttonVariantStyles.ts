import { CustomTheme as Theme } from '../../../../types';
import { ButtonVariant, OutlinedColors, SolidColors } from '../constants/types';

export const buttonVariantStyles = (
  variant: ButtonVariant,
  theme: Theme,
  color: SolidColors | OutlinedColors
) =>
  ({
    solid: `
      text-transform: uppercase;
      border: ${theme.spacing.none};

      &:focus {
        box-shadow: 0px 2px 1px 0px #1A1A1A0D, 0px 0px 0px 4px ${theme.colors.blue_10};
        background-color: ${theme.palette[color]};
      }

      &:disabled {
        opacity: 38%;
      }
  `,
    outlined: `
      border: 1px solid;
      text-transform: uppercase;
        ${
          color === 'neutral'
            ? theme.palette.neutralDark_100
            : theme.palette[color]
        };

      &:disabled {
        color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        background-color: ${theme.colors.white};
        border-color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        cursor: default;
      }
    `,
    text: `
      border: none;

      &:disabled {
        color: ${theme.palette.neutralLight_100};
      }
    `,
  }[variant]);
