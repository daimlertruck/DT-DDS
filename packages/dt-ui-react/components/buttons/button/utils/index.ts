export * from './buttonVariantColors';
export * from './buttonVariantStyles';
