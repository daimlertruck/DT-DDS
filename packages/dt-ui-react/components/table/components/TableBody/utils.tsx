import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
