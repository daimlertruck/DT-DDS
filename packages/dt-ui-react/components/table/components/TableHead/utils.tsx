import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
