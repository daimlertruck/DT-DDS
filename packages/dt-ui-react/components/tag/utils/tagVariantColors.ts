import { CustomTheme as Theme } from '../../../types';
import { TagColor, TagVariant } from '../constants';

interface TagVariantColors {
  border: string | number;
  color: string;
  borderColor?: string;
  backgroundColor: string;
  '&:hover'?: {
    borderColor?: string;
    backgroundColor: string;
    color: string;
  };
}

interface TagVariantColorsProps {
  theme: Theme;
  color: TagColor;
  variant: TagVariant;
  hasHover: boolean;
  isDisabled: boolean;
}

const getColorWithShade = ({
  theme,
  color,
  shade,
}: Pick<TagVariantColorsProps, 'color'> & {
  shade: '90' | '60' | '30' | '10';
  theme: Theme;
}) => {
  const { colors, palette } = theme;

  if (color === 'primary') {
    return {
      '90': palette.primaryDark_500,
      '60': palette.primaryDark_200,
      '30': palette.primaryLight_100,
      '10': palette.primaryLight_400,
    }[shade];
  }

  const colorWithShade: keyof Theme['colors'] = `${color}_${shade}`;

  return colors[colorWithShade];
};

export const tagVariantColors = ({
  theme,
  color,
  variant,
  hasHover,
  isDisabled,
}: TagVariantColorsProps): TagVariantColors => {
  const colorShade90 = getColorWithShade({
    theme,
    color,
    shade: '90',
  });
  const colorShade60 = getColorWithShade({
    theme,
    color,
    shade: '60',
  });
  const colorShade30 = getColorWithShade({
    theme,
    color,
    shade: '30',
  });
  const colorShade10 = getColorWithShade({
    theme,
    color,
    shade: '10',
  });

  const colorWithDisabled = isDisabled ? colorShade30 : colorShade60;

  return {
    solid: {
      border: theme.spacing.none,
      color: theme.colors.white,
      backgroundColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          backgroundColor: colorShade90,
          color: theme.colors.white,
        },
      }),
    },
    outlined: {
      border: `1px solid ${colorShade60}`,
      color: colorWithDisabled,
      backgroundColor: colorShade10,
      borderColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          borderColor: colorShade90,
          backgroundColor: colorShade10,
          color: colorShade90,
        },
      }),
    },
  }[variant];
};
