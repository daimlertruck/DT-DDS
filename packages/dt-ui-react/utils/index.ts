export { getTextStyles } from './typography';
export * from './testingUtils';
export { default as useClickOutside } from './useClickOutside';
export { default as debounce } from './debounce';
export { acronymGenerator } from './acronymGenerator';
export * from './constants';
export { hexToRgba } from './hexToRgba';
