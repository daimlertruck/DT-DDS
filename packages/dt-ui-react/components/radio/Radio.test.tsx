import { render, fireEvent, screen } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../utils';

import { Radio, RadioGroup } from '.';

describe('<Radio /> component', () => {
  const ProvidedRadioGroup = withProviders(RadioGroup);
  it('renders Radio group with Radio 1 checked and Radio 3 disabled', () => {
    const { container } = render(
      <ProvidedRadioGroup onChange={() => null} name='radio-group-name'>
        <Radio
          value='1'
          isDisabled={false}
          label='Label 1'
          isDefaultChecked={true}
        />
        <Radio value='2' isDisabled={false} label='Label 2' />
        <Radio value='3' isDisabled={true} label='Label 3' />
      </ProvidedRadioGroup>
    );
    expect(container).toMatchSnapshot();
  });

  it('changes selected Radio', () => {
    render(
      <ProvidedRadioGroup onChange={() => null} name='radio-group-name'>
        <Radio value='1' label='Label 1' />
        <Radio value='2' label='Label 2' isDefaultChecked={true} />
      </ProvidedRadioGroup>
    );

    fireEvent.click(screen.getByText('Label 1'));
    expect(screen.getByLabelText('Label 1')).toBeChecked();
  });
});
