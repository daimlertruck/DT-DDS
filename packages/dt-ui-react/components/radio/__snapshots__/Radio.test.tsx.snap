// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> component renders Radio group with Radio 1 checked and Radio 3 disabled 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 12px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  background: #FFFFFF;
  border: 1px solid #D3D4D5;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.emotion-4:disabled {
  background: #F8F9FA;
  pointer-events: none;
}

.emotion-4:disabled+label {
  color: #BABBBB;
}

.emotion-4:checked {
  border-width: 5px;
  border-color: #008DFC;
}

.emotion-4:checked:disabled {
  border-color: #D3D4D5;
}

.emotion-4:checked:before {
  content: '';
  height: 10px;
  width: 10px;
  border-radius: 50%;
  display: block;
}

.emotion-6 {
  font-weight: 400;
  font-size: 18px;
  line-height: 24.12px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #7C7C7D;
  cursor: inherit;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="radio-group"
    direction="row"
  >
    <div
      class="emotion-2 emotion-3"
      style="flex-direction: row; gap: 8px; cursor: pointer;"
    >
      <input
        checked=""
        class="emotion-4 emotion-5"
        id="1-id"
        name="radio-group-name"
        type="radio"
        value="1"
      />
      <label
        class="emotion-6 emotion-7"
        for="1-id"
      >
        Label 1
      </label>
    </div>
    <div
      class="emotion-2 emotion-3"
      style="flex-direction: row; gap: 8px; cursor: pointer;"
    >
      <input
        class="emotion-4 emotion-5"
        id="2-id"
        name="radio-group-name"
        type="radio"
        value="2"
      />
      <label
        class="emotion-6 emotion-7"
        for="2-id"
      >
        Label 2
      </label>
    </div>
    <div
      class="emotion-2 emotion-3"
      style="flex-direction: row; gap: 8px; cursor: default;"
    >
      <input
        class="emotion-4 emotion-5"
        disabled=""
        id="3-id"
        name="radio-group-name"
        type="radio"
        value="3"
      />
      <label
        class="emotion-6 emotion-7"
        for="3-id"
      >
        Label 3
      </label>
    </div>
  </div>
</div>
`;
